### ISSUE TITLE
Microsoft broke your extension

### ISSUE THREAD

**Issue by CregoryLC** [zach-adams/simple-new-tab-url#14]
In Edge on Windows 11 you can no longer set the sites or permissions for this extension. As a result, setting the new tab URL to a local URL (example: file:///E:/home.html) causes the extension to open a blank page. At least the blank page is still better than Bing or the default crappy Edge new tab. I have edge open file:///E:/home.html on startup and have it set as the home page with the home button shown on the tool bar. So, opening the local URL is a 2-step task: open a new tab that is broken/blank the click on the home button. I imagine before too long we won't need a keyboard or mouse. We'll turn our computers on and Microsux will just operate it for us and we will just get what they decide to give us.

**Comment by resolval** [comment]
Same issue - I was using this extension for opening a local file on new tabs - it stopped working with only a black screen so I've just removed it and reinstalled the latest and when I open a new tab now I get the message: "If your URL is a local file then you will have to enable the "Allow access to file URLs" permission for this extension:" the issue is there is no longer an option for that on the extension settings.

Why Microsoft have decided we can no longer decide ourselves is beyond me - is there a work around / fix that can solve this at all?

**Comment by resolval** [comment]
Quick follow-up - I've just installed another new tab extension and that still has the "Allow access to file URLs" so not sure why it's not showing in the options for this one.

I'd prefer to use this extension as it doesn't leave the URL highlighted and selected on a new tab.

**Comment by zach-adams** [comment]
Just a quick update, I think this issue might be related to the new "auto-highlight/select URL" option I added and some weird permissions issues with local file URLs. I'm researching a solution, here's a TL;DR for the meantime:
**TL;DR: You can try enabling the "auto-highlight" option and that _should_ let the redirect work for the time being on local file URLs while I work out these weird permissions issue. Or you could run a little web-server program like [hfs.exe](https://www.rejetto.com/hfs/) in the background to load your local file over http at a URL like "http://localhost:5000/file.html" to avoid the permissions issue entirely.**

To explain further: that blank page you're seeing is probably the "redirect.html" page from the extension trying to redirect itself to your local file URL without auto-highlighting the URL, and hitting an uncaught permissions error. Opening a new tab to your chosen file URL with the local file URL _not_ being auto-highlighted is a separate action/event from opening it _with_ it being auto-highlighted, and that action/event requires more permissions beyond the "Allow access to file URLs" toggle which the auto-highlighting one requires. So far I've found that it seemingly requires full access to all the content to any of your local files loaded in the browser (as opposed to just the file URLs), and those extremely broad & unnecessary permissions most people would rightly find quite ludicrous for just a simple new tab URL extension.  

So I'm trying to find a way to lessen the blow of that permissions request by only requesting those extra permissions when it becomes necessary for that specific "local file URL without auto-highlighting" case (and explaining why in a user-friendly way) or just finding a way to bypass it altogether.

**Comment by zach-adams** [comment]
Oh on top of all of that, those extra permissions are handled slightly differently across both Chrome & Edge browsers, as well as their separate webstores, so it's not consistent and I need to know why that is.
I'll focus on fixing Edge first since it has the most users then wrap back around to Chrome

**Comment by resolval** [comment]
Hi

Firstly thank you for this extension and all the work that goes into it!

I've tried enabling the auto-highlight but this still results in the : "If your URL is a local file then you will have to enable the "Allow access to file URLs" permission for this extension:" page.  (I did close and restart the browser just in case as well).

Colin
